test/helper/launcher/launcher.go

```diff
@@ -0,0 +1,210 @@
+package launcher
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+)
+
+const (
+	ExpectedContext = "kind-kind"
+
+	LauncherTestInstall = "test-ako"
+
+	HelmRepoURL = "https://mongodb.github.io/helm-chart"
+
+	RepoRef = "mongodb"
+
+	OperatorChart = "mongodb-atlas-operator"
+
+	AtlasURI = "https://cloud-qa.mongodb.com"
+)
+
+type Launcher struct {
+	credentials   AtlasCredentials
+	version       string
+	cmdOutput     bool
+	clearKind     bool
+	clearOperator bool
+}
+
+// NewLauncher creates a new operator Launcher
+func NewLauncher(creds AtlasCredentials, version string, cmdOutput bool) *Launcher {
+	return &Launcher{credentials: creds, version: version, cmdOutput: cmdOutput}
+}
+
+// NewFromEnv creates an operator Launcher using defaults and credentials from env vars
+func NewFromEnv(version string) *Launcher {
+	return NewLauncher(credentialsFromEnv(), version, true)
+}
+
+// MustSetEnv sets the env var value given, or panics if the env var is not set
+func MustSetEnv(envvar string) string {
+	value, ok := os.LookupEnv(envvar)
+	if !ok {
+		panic(fmt.Errorf("environment variable %s not set", envvar))
+	}
+	return value
+}
+
+// Launch will try to launch the operator and apply the given YAML for it to handle
+func (l *Launcher) Launch(yml string, waitCfg *WaitConfig) error {
+	if err := l.ensureK8sCluster(); err != nil {
+		return err
```

Comment:
wrap all errors below. In this case:
```suggestion
		return fmt.Errorf("error ensuring k8s cluster: %w", err)
```